### PR TITLE
chore: upgrade polkadot to v0.9.36

### PR DESCRIPTION
**Pull request by trusch**
## fixes KILTProtocol/ticket#2351

Upgrade polkadot version to 0.9.36

### Notable changes in 0.9.32 -> 0.9.33

[Full changelog](https://github.com/paritytech/polkadot/releases/tag/v0.9.33)

This release is super small and only consists of removing the sp_tasks::spawn api, that wasn't used by us or any others (https://github.com/paritytech/substrate/pull/12639). The only other change was an internal refactoring of the state database in https://github.com/paritytech/substrate/pull/12239

### Notable changes in 0.9.33 -> 0.9.34

[Full changelog](https://github.com/paritytech/polkadot/releases/tag/v0.9.34)

This is a bigger runtime only release mainly about improvements to opengov and collectives. The only notable change that required code changes was the removal of the wasmtime feature flag in some crates https://github.com/paritytech/substrate/pull/12684

### Notable changes in 0.9.34 -> 0.9.35

[Full changelog](https://github.com/paritytech/polkadot/releases/tag/v0.9.35)

Most notable are the two changes https://github.com/paritytech/substrate/pull/12891
and https://github.com/paritytech/substrate/pull/12868. Also there is now the possibility to use a new RPC server that supports http and websocket requests on the same server. We should look into this since this will eventually cause the older severs to be deprecated in the future (https://github.com/paritytech/substrate/pull/12663). In https://github.com/paritytech/substrate/pull/12485 a general message queue pallet was introduced that might be helpful for receivng XCM in the future.

### Notable changes in 0.9.35 -> 0.9.36

[Full changelog](https://github.com/paritytech/polkadot/releases/tag/v0.9.36)

This is a hotfix release to address an issue in the dispute coordinator on the node level.

## How to test:
Please provide a brief step-by-step instruction.
If necessary provide information about dependencies (specific configuration, branches, database dumps, etc.)

- Step 1
- Step 2
- etc.

## Checklist:

- [ ] I have verified that the code works
  - [ ] No panics! (checked arithmetic ops, no indexing `array[3]` use `get(3)`, ...)
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: runtimes/spiritnet/src/lib.rs [hunk 1]
```diff
@@ -1404,21 +1404,21 @@ impl_runtime_apis! {
...
 			log::info!("try-runtime::on_runtime_upgrade spiritnet.");
-			let weight = Executive::try_runtime_upgrade().unwrap();
+			let weight = Executive::try_runtime_upgrade(checks).unwrap();
 			(weight, BlockWeights::get().max_block)
 		}
 
-		fn execute_block(block: Block, state_root_check: bool, select: frame_try_runtime::TryStateSelect) -> Weight {
+		fn execute_block(block: Block, state_root_check: bool, sig_check: bool, select: frame_try_runtime::TryStateSelect) -> Weight {
 			log::info!(
 				target: "runtime::spiritnet", "try-runtime: executing block #{} ({:?}) / root checks: {:?} / sanity-checks: {:?}",
```
Comment (ntn-x2) [inline runtimes/spiritnet/src/lib.rs:1415]: I guess we also want to print the signature checks flag here as well?

File: runtimes/standalone/src/lib.rs [hunk 2]
```diff
@@ -1123,21 +1123,21 @@ impl_runtime_apis! {
...
 			log::info!("try-runtime::on_runtime_upgrade mashnet-node standalone.");
-			let weight = Executive::try_runtime_upgrade().unwrap();
+			let weight = Executive::try_runtime_upgrade(checks).unwrap();
 			(weight, runtime_common::BlockWeights::get().max_block)
 		}
 
-		fn execute_block(block: Block, state_root_check: bool, select: frame_try_runtime::TryStateSelect) -> Weight {
+		fn execute_block(block: Block, state_root_check: bool, sig_check: bool, select: frame_try_runtime::TryStateSelect) -> Weight {
 			log::info!(
 				target: "runtime::mashnet_node_standalone", "try-runtime: executing block #{} ({:?}) / root checks: {:?} / sanity-checks: {:?}",
```
Comment (ntn-x2) [inline runtimes/standalone/src/lib.rs:1134]: Here as well?

**Comment by ntn-x2** [comment]
@trusch as an exercise, but also for future migrations PRs, would you mind to change the PR description to follow the traces of what William used to do for migration PRs? E.g., https://github.com/KILTprotocol/kilt-node/pull/434. This helps finding out potential stuff that has changed to you, while going over the changes, and to us, while reviewing the PR. There has already been few cases where parachains have stalled because of some missed migrations or storage changes 😋 I did check that there was no relevant migration for us, but for the sake of traceability, it would be good to have a good PR description for upgrades, especially when updating 4 versions at once 🙏🙏🙏

File: nodes/standalone/Cargo.toml [hunk 3]
```diff
@@ -23,6 +23,7 @@ substrate-build-script-utils.workspace = true
 clap = {workspace = true, features = ["derive"]}
 jsonrpsee = {workspace = true, features = ["server"]}
 hex-literal.workspace = true
+futures = { version = "0.3.21", features = ["thread-pool"]}
```
[the remark this answers is not in the document]
Comment (trusch) [reply nodes/standalone/Cargo.toml]: done

File: runtimes/spiritnet/src/lib.rs (hunk 1 above)
Comment (trusch) [reply runtimes/spiritnet/src/lib.rs]: done

File: runtimes/standalone/src/lib.rs (hunk 2 above)
Comment (trusch) [reply runtimes/standalone/src/lib.rs]: done

File: runtimes/clone/src/lib.rs [hunk 4]
```diff
@@ -697,10 +697,11 @@ impl_runtime_apis! {
 
 		fn execute_block(block: Block, state_root_check: bool, sig_check: bool, select: frame_try_runtime::TryStateSelect) -> Weight {
 			log::info!(
-				target: "runtime::clone", "try-runtime: executing block #{} ({:?}) / root checks: {:?} / sanity-checks: {:?}",
+				target: "runtime::spiritnet", "try-runtime: executing block #{} ({:?}) / root checks: {:?} / sig check: {:?} / sanity-checks: {:?}",
```
Comment (ntn-x2) [inline runtimes/clone/src/lib.rs:700]: This is the clone runtime, so clone is the right name 😄

File: runtimes/peregrine/src/lib.rs [hunk 5]
```diff
@@ -1418,11 +1418,11 @@ impl_runtime_apis! {
 
 		fn execute_block(block: Block, state_root_check: bool, signature_check: bool, select: frame_try_runtime::TryStateSelect) -> Weight {
 			log::info!(
-				target: "runtime::peregrine", "try-runtime: executing block #{} ({:?}) / root checks: {:?} / signature checks: {:?} sanity-checks: {:?}",
+				target: "runtime::spiritnet", "try-runtime: executing block #{} ({:?}) / root checks: {:?} / sig check: {:?} / sanity-checks: {:?}",
```
Comment (ntn-x2) [inline runtimes/peregrine/src/lib.rs:1421]: This is the peregrine runtime.

File: runtimes/standalone/src/lib.rs [hunk 6]
```diff
@@ -1131,10 +1131,11 @@ impl_runtime_apis! {
 
 		fn execute_block(block: Block, state_root_check: bool, sig_check: bool, select: frame_try_runtime::TryStateSelect) -> Weight {
 			log::info!(
-				target: "runtime::mashnet_node_standalone", "try-runtime: executing block #{} ({:?}) / root checks: {:?} / sanity-checks: {:?}",
+				target: "runtime::spiritnet", "try-runtime: executing block #{} ({:?}) / root checks: {:?} / sig check: {:?} / sanity-checks: {:?}",
```
Comment (ntn-x2) [inline runtimes/standalone/src/lib.rs:1134]: Standalone runtime.

File: runtimes/clone/src/lib.rs (hunk 4 above)
Comment (trusch) [reply runtimes/clone/src/lib.rs]: Thanks, stupid copy-paste error.

File: runtimes/peregrine/src/lib.rs (hunk 5 above)
Comment (trusch) [reply runtimes/peregrine/src/lib.rs]: done

File: runtimes/standalone/src/lib.rs (hunk 6 above)
Comment (trusch) [reply runtimes/standalone/src/lib.rs]: fixed

**Review by weichweich — APPROVED** [review]
LGTM

I would love to create a diff of our metadata for this release (because of the extrinsic indices). But that'll be another task.

**Comment by ntn-x2** [comment]
> I would love to create a diff of our metadata for this release (because of the extrinsic indices). But that'll be another task.

This will be done anyway in the SDK when we re-run type augmentation. But anyway it could be part of https://github.com/KILTprotocol/ticket/issues/2241 or at least related to it.

**Pull request merged**